src/widgets/helper/ChannelView.cpp

```diff
@@ -1402,6 +1402,13 @@ MessageElementFlags ChannelView::getFlags() const
         {
             split = dynamic_cast<Split *>(searchPopup->parentWidget());
         }
+
+        auto *userInfoPopup = dynamic_cast<UserInfoPopup *>(
+            this->parentWidget()->parentWidget()->parentWidget());
```

Comment:
I'm here to let you know I strongly dislike this solution. I know that we use this all over the place so it can stay, however I will not be silent.

Also what if ChannelView's parent does not have a parent? This segfaults right? I think we need to check the entire painful chain for objects we expect.

Reply:
yeah I'm not super fond of this either, I'd rather we find some other solution to this that doesn't involve digging through 3 sets of parents, or relying on the original channels "mod view" state.
maybe a mod view button in the userchannelpopup itself to control whether or not it should be visible?

Reply:
We could have a global mod state for usercards and if it's opened in the context of a moderated channel it could show some moderation buttons, ideally excluding ban and timeout ones, so they wouldn't duplicate the ones already available in the usercard.

Reply:
> Also what if ChannelView's parent does not have a parent? This segfaults right?

This segfaults when opening the overlay window right now.

Reply:
I'll make it check if each parentWidget() is a nullptr. That should should solve it right?

Reply:
This should be solved now.